code/modules/clothing/shoes/clownshoes.dm

```diff
@@ -0,0 +1,204 @@
+/obj/item/clothing/shoes/clown_shoes
+	desc = "The prankster's standard-issue clowning shoes. Damn, they're huge! Ctrl-click to toggle waddle dampeners."
+	name = "clown shoes"
+	icon_state = "clown"
+	inhand_icon_state = "clown_shoes"
+	slowdown = SHOES_SLOWDOWN+1
+	pocket_storage_component_path = /datum/component/storage/concrete/pockets/shoes/clown
+	var/enabled_waddle = TRUE
+	lace_time = 20 SECONDS // how the hell do these laces even work??
+
+/obj/item/clothing/shoes/clown_shoes/Initialize()
+	. = ..()
+	AddComponent(/datum/component/squeak, list('sound/effects/clownstep1.ogg'=1,'sound/effects/clownstep2.ogg'=1), 50)
+
+/obj/item/clothing/shoes/clown_shoes/equipped(mob/user, slot)
+	. = ..()
+	if(slot == ITEM_SLOT_FEET)
+		if(enabled_waddle)
+			user.AddElement(/datum/element/waddling)
+		if(user.mind && user.mind.assigned_role == "Clown")
+			SEND_SIGNAL(user, COMSIG_ADD_MOOD_EVENT, "clownshoes", /datum/mood_event/clownshoes)
+
+/obj/item/clothing/shoes/clown_shoes/dropped(mob/user)
+	. = ..()
+	user.RemoveElement(/datum/element/waddling)
+	if(user.mind && user.mind.assigned_role == "Clown")
+		SEND_SIGNAL(user, COMSIG_CLEAR_MOOD_EVENT, "clownshoes")
+
+/obj/item/clothing/shoes/clown_shoes/CtrlClick(mob/living/user)
+	if(!isliving(user))
+		return
+	if(user.get_active_held_item() != src)
+		to_chat(user, "<span class='warning'>You must hold the [src] in your hand to do this!</span>")
+		return
+	if (!enabled_waddle)
+		to_chat(user, "<span class='notice'>You switch off the waddle dampeners!</span>")
+		enabled_waddle = TRUE
+	else
+		to_chat(user, "<span class='notice'>You switch on the waddle dampeners!</span>")
+		enabled_waddle = FALSE
+
+
+
+/obj/item/clothing/shoes/clown_shoes/sec
+	desc = "The law *IS* a joking matter! Ctrl-click to toggle waddle dampeners."
+	name = "clown jacks"
+	icon_state = "clown_sec"
+	inhand_icon_state = "clown_shoes_sec"
+	slowdown = SHOES_SLOWDOWN
+	pocket_storage_component_path = /datum/component/storage/concrete/pockets/shoes
+
+/obj/item/clothing/shoes/clown_shoes/sec/Initialize()
+	. = ..()
+	AddComponent(/datum/component/squeak, list('sound/effects/footstep/wood5.ogg'=1), 100, 100, 0)
+
+
+/obj/item/clothing/shoes/clown_shoes/prison
+	desc = "That trial was a joke! A kangaroo court!  Ctrl-click to toggle waddle dampeners."
```

Comment:
would it be possible to add cuffed and not-cuffed version of these prison shoes, in case you wanna show off your prison rep as a clown but don't want to move at an ungodly slow pace?

Reply:
Yes but it would involve a lot of copypaste from the orange shoe file, or some arcane legcuff component beyond my abilities.